Tests/FluentAssertions.Specs/Equivalency/DataEquivalencySpecs.cs

```diff
@@ -0,0 +1,4578 @@
+        private static void ApplyChange(DataColumnCollection columns, ChangeType changeType)
+        {
+            switch (changeType)
+            {
+                case ChangeType.Added:
+                    columns.Add("Test", typeof(int));
+                    break;
+                case ChangeType.Changed:
+                    columns[1].ColumnName += "different";
+                    break;
+                case ChangeType.Removed:
+                    columns.RemoveAt(1);
+                    break;
+            }
+        }
+
+        [SuppressMessage("Style", "IDE0010:Add missing cases", Justification = "Every enum value is covered")]
+        private static void ApplyChange(PropertyCollection extendedProperties, ChangeType changeType)
+        {
+            switch (changeType)
+            {
+                case ChangeType.Added:
+                    extendedProperties[Guid.NewGuid()] = Guid.NewGuid();
+                    break;
+                case ChangeType.Changed:
+                    extendedProperties[extendedProperties.Keys.Cast<object>().First()] = Guid.NewGuid();
+                    break;
+                case ChangeType.Removed:
+                    extendedProperties.Remove(extendedProperties.Keys.Cast<object>().First());
+                    break;
+            }
+        }
+
+        [SuppressMessage("Style", "IDE0010:Add missing cases", Justification = "Every enum value is covered")]
+        private static void ApplyChange(ConstraintCollection constraints, DataColumn columnForNewConstraint, ChangeType changeType)
+        {
+            switch (changeType)
+            {
+                case ChangeType.Added:
+                    constraints.Add(
+                        new UniqueConstraint(
+                            "Test",
+                            columnForNewConstraint));
+                    break;
+                case ChangeType.Changed:
+                    constraints[1].ConstraintName += "different";
+                    break;
+                case ChangeType.Removed:
+                    constraints.RemoveAt(1);
+                    break;
+            }
+        }
+
+        [SuppressMessage("Style", "IDE0010:Add missing cases", Justification = "Every enum value is covered")]
+        private static void ApplyChange(DataRowCollection rows, DataTable dataTable, ChangeType changeType)
+        {
+            switch (changeType)
+            {
+                case ChangeType.Added:
+                    InsertDummyRow(dataTable);
+                    break;
+                case ChangeType.Changed:
+                    rows[1]["String"] += "different";
+                    break;
+                case ChangeType.Removed:
+                    rows.RemoveAt(0);
+                    break;
+            }
+        }
+
+        public class DataSetEquivalencySpecs : DataEquivalencySpecs
+        {
+            [Fact]
+            public void When_DataSets_are_identical_it_should_succeed()
```

Comment:
❌ Please don't use literal name of code elements. Try to keep the test names functional.

Reply:
Can you provide more detail here? For instance, in most of the new equivalency step implementations, the code under test must explicitly compare specific properties, and tests validating these comparisons are named after the properties so that it is clear that they are covering it. Are you saying that these tests should _not_ specifically state what it is that they are testing?? Wouldn't that be more confusing? I feel like I'm not understanding something here. :-)

Reply:
In general, we prefer to keep names more functional, without referring to the exact names of code elements, simply because they tend to change. In this case, the problem isn't that big, since `DataSet` is never going to change. But I would still prefer to name it something like `When_data_sets_are_indentical_it_should_succeed`.

Reply:
Sometimes it is just a matter of rephrasing a bit, like `When_data_sets_are_indentical_it_should_succeed`. Of course, in this particular case, `DataSet` will never change, but I see people do it with their code elements. And to keep the code-base as consistent as possible and an example to other developers, it would be very nice of you to follow it ;-)

Reply:
I'm feeling conflicted on this. I'll make whatever change ultimately you indicate, but my _feeling_ on it is, in this case, that the test is _very directly_ testing specific types and members, and it obfuscates things to use a "human-readable description" of the subject instead. Should I be doing a similar transformation for properties? For instance, should `When_AllowDBNull_does_not_match_and_property_is_not_excluded_it_should_fail` be something like `When_allow_database_nulls_property_does_not_match`? Or perhaps `When_the_property_that_means_allow_database_nulls_does_not_match`?

In the cases of collections and dictionaries, there _is no_ canonical type; the test matches _anything_ that implements `ICollection` or `IDictionary<TKey, TValue>` or what have you. So, the idea of calling it `a_collection` or `a_dictionary` seems quite sensible. But, there is no interface for `DataSet` objects -- they will _always_ be of type `DataSet`, or a type directly derived from `DataSet` specifically.

In these tests, if the subject of the test is called out as a particular type or property in the test name, it's not that the test is testing "a thing like" that, it is _by definition_ testing _exactly that thing_. Most of the tests of the System.Data equivalency code are like that, because the heavy interconnectedness of these types, with all manner of cyclical references, meant that I wrote the equivalency check so that it explicitly tests specific properties. Consider the test:

`When_Rows_do_not_match_and_property_is_excluded_it_should_succeed`

That test name is _very explicitly_ testing what happens if you call `options.Excluding(dataTable => dataTable.Rows)`. If the test name doesn't contain the word `Rows`, then it isn't being clear about what it's doing any more.

Do you still think that this change is called-for? If so, please tell me exactly which changes to make, because I don't have my head wrapped around it in such a way as to be able to intuit for myself what they are. :-) E.g.:

* Should all type names be replaced by a human description?
* If a test is about what happens when you exclude a _specific property_, how should the test name indicate that? Or, should it not indicate it?

Let me know what to do and I'll make the corresponding change. Thanks :-)

Reply:
OK. I think your point makes sense. And we can also improve naming afterwards.